### DISCUSSION TITLE
localstack-starter

### DISCUSSION THREAD

**Discussion by alikian** [awspring/spring-cloud-aws#859] [Ideas]
Localstack is a key component to local development and testing. With awsspring 3.x release bootstrap.yaml is gone and there will be issues for migrating from older version of spring.
I am suggesting for have a module call **locastack-starter**, this module will do followings:

1. Start localstack based on spring profiles like dev or test
2. Create aws clients like secretmanager, dynamodb, sqs using localstack region and endpoint override
3. Run cloudformation template to create resources 
4. Seed data to resources like Secret Manager, DynamodDB
5. Stop localstack and the end

**Comment by MatejNedic** [comment]
Hey @alikian , It is already planned in scope of https://github.com/awspring/spring-cloud-aws/issues/793 .
Please comment on that issue if you have more ideas to add.
